lib/new_relic/agent/log_event_attributes.rb

```diff
@@ -0,0 +1,123 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    class LogEventAttributes
+      MAX_ATTRIBUTE_COUNT = 240 # limit is 255, assume we send 15
+      ATTRIBUTE_KEY_CHARACTER_LIMIT = 255
+      ATTRIBUTE_VALUE_CHARACTER_LIMIT = 4094
+
+      attr_reader :custom_attributes, :already_warned_custom_attribute_count_limit
```

Comment:
The naming for `already_warned_custom_attribute_count_limit` kinda leads me to think there's just a warning about reaching a limit but it's actually making a cut.  Could be `custom_attribute_count_limit_reached`? Unless the important takeaway is that we sent a warning off.

Reply:
Renamed here: 8fe40f1ad13aab82fc488efe0bf0f0bf7629b0e2 
I took out "count" because it didn't seem necessary to me, but it does match the constant name more closely. Let me know if you'd like me to put it back in.